pexv2/presentation_exchange.go

```diff
@@ -0,0 +1,170 @@
+package pexv2
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/tbd54566975/web5-go/vc"
+	jsonschema "github.com/xeipuuv/gojsonschema"
+)
+
+type tokenPath struct {
+	Token string
+	Paths []string
+}
+
+// SelectCredentials selects vcJWTs based on the constraints defined in the presentation definition
+func SelectCredentials(vcJWTs []string, pd PresentationDefinition) ([]string, error) {
+
+	result := make([]string, 0)
+	for _, inputDescriptor := range pd.InputDescriptors {
+		matchedVcJWTs, err := selectCredentialsPerInputDescriptor(vcJWTs, inputDescriptor)
+		if err != nil {
+			return []string{}, err
+		}
+		if len(matchedVcJWTs) == 0 {
+			return []string{}, nil
+		}
+		result = append(result, matchedVcJWTs...)
+
+	}
+
+	result = dedupeResult(result)
+	return result, nil
+}
+
+func dedupeResult(input []string) []string {
+	sort.Strings(input)
+	var result []string
+
+	for i, item := range input {
+		if i == 0 || input[i-1] != item {
+			result = append(result, item)
+		}
+	}
+	return result
+}
+
+func selectCredentialsPerInputDescriptor(vcJWTs []string, inputDescriptor InputDescriptor) ([]string, error) {
+	answer := make([]string, 0)
+	tokenizedField := make([]tokenPath, 0)
```

Comment:
Naming: How do you feel about changing the name of `tokenPath` type to `tokenizedField`, and this variable to be called `tokenizedFields` plural?